dialogue-blocking-channels/src/main/java/com/palantir/dialogue/blocking/DefaultCallingThreadExecutor.java

```diff
@@ -54,49 +52,44 @@
     }
 
     @Override
-    public ListenableFuture<?> submit(Runnable task) {
-        return queue.submit(task);
+    public void submit(Runnable task) {
+        queue.submit(task);
     }
 
     @Override
     public void executeQueue(ListenableFuture<?> await) {
         Preconditions.checkState(Thread.currentThread().getId() == threadId, "Executing queue on different thread");
-        await.addListener(() -> queue.submitNotifier(queue::poison), DialogueFutures.safeDirectExecutor());
+        DialogueFutures.addDirectListener(await, () -> queue.submitNotifier(queue::poison));
         try {
-            RunnableFuture<?> toRun;
+            Runnable toRun;
             while ((toRun = queue.getWork()) != null) {
-                toRun.run();
+                try {
+                    toRun.run();
+                } catch (Throwable t) {
+                    log.error("Failed to execute runnable {}", toRun, t);
```

Comment:
This should basically not happen, so I think we can have this here.

Reply:
Exactly, we submit a task which uses a SettableFuture as output rather than throwing exceptions outside of the Runnable.